### PR TITLE
Adopt CIRCL for HPKE implementation

### PR DESCRIPTION
**Pull request by chris-wood**
cc @bwesterb, as this is the first step towards adopting the xyber-based suite.

### CODE REVIEW

File: ohttp.go [hunk 1]
```diff
@@ -372,10 +370,12 @@ func (c Client) EncapsulateRequest(request []byte) (EncapsulatedRequest, Encapsu
...
 func (c EncapsulatedRequestContext) DecapsulateResponse(response EncapsulatedResponse) ([]byte, error) {
 	// secret = context.Export("message/bhttp response", Nk)
-	secret := c.context.Export(c.responseLabel, c.suite.AEAD.KeySize())
+	_, KDF, AEAD := c.suite.Params()
+
+	secret := c.context.Export(c.responseLabel, AEAD.KeySize())
 
 	// response_nonce = random(max(Nn, Nk)), taken from the encapsualted response
-	responseNonceLen := max(c.suite.AEAD.KeySize(), c.suite.AEAD.NonceSize())
+	responseNonceLen := max(int(AEAD.KeySize()), 12)
```
Comment (chris-wood) [inline ohttp.go:378]: @armfazh, can we get a function in CIRCL that returns the nonce length for an AEAD?

Comment (armfazh) [reply ohttp.go:378]: https://github.com/cloudflare/circl/pull/424

does this help?

Comment (chris-wood) [reply ohttp.go:378]: Yes, perfect!

Comment (armfazh) [reply ohttp.go:378]: done

**Review by bwesterb — APPROVED** [review]

**Pull request merged**